pkg/build/strategies/strategies.go

```diff
@@ -6,27 +6,40 @@ import (
 	"github.com/openshift/source-to-image/pkg/build/strategies/onbuild"
 	"github.com/openshift/source-to-image/pkg/build/strategies/sti"
 	"github.com/openshift/source-to-image/pkg/docker"
+	"github.com/openshift/source-to-image/pkg/util/updater"
 )
 
 // GetStrategy decides what build strategy will be used for the STI build.
 // TODO: deprecated, use Strategy() instead
-func GetStrategy(config *api.Config) (build.Builder, error) {
+func GetStrategy(config *api.Config) (build.Builder, []*api.BuildStepInfo, error) {
 	return Strategy(config, build.Overrides{})
 }
 
 // Strategy creates the appropriate build strategy for the provided config, using
 // the overrides provided. Not all strategies support all overrides.
-func Strategy(config *api.Config, overrides build.Overrides) (build.Builder, error) {
+func Strategy(config *api.Config, overrides build.Overrides) (build.Builder, []*api.BuildStepInfo, error) {
+	var builder build.Builder
+	var buildInfo []*api.BuildStepInfo
+	var err error
+
 	image, err := docker.GetBuilderImage(config)
 	if err != nil {
-		return nil, err
+		buildInfo = updater.UpdateFailureReason(buildInfo, api.ReasonGenericS2IBuildFailed)
```

Comment:
at least a generic "docker interaction" sort of error if "failed getting builder image" is not appropriate